airvantage.go

```diff
@@ -101,8 +105,69 @@ type apiError struct {
 func (av *AirVantage) parseResponse(resp *http.Response, respStruct interface{}) error {
 	defer resp.Body.Close()
 
+	if err := av.parseError(resp); err != nil {
+		return err
+	}
+
+	if respStruct == nil {
+		return nil
+	}
+
+	var payload io.Reader = resp.Body
+	if av.Debug {
+		body, err := io.ReadAll(resp.Body)
+		if err != nil {
+			return err
+		}
+		av.log.Printf("Path: %s\nContent: %s\n", resp.Request.URL, string(body))
+		payload = bytes.NewReader(body)
+	}
+
+	if err := json.NewDecoder(payload).Decode(respStruct); err != nil {
+		return fmt.Errorf("unable to parse API response: %s", err)
+	}
+
+	return nil
+}
+
+// parseResponseSystemSecurityInfo is similar to parseResponse
+// since the response is Java object serialized we have to remove these references
+// respStruct must be a pointer to a struct where the JSON will be deserialized.
+func (av *AirVantage) parseResponseSystemSecurityInfo(resp *http.Response, respStruct interface{}) error {
+	defer resp.Body.Close()
+
+	err := av.parseError(resp)
+	if err != nil {
+		return err
+	}
+
+	if respStruct == nil {
```

Comment:
Same here, you can save lines writting: `if err != nil || respStruc == nil`

Reply:
You mean something like:
```
if err != nil || respStruc == nil {
  return err
}
```

It would indeed be functionaly equivalent, but don't enhance readability imho since you need to think to actually understand the returned value in both cases

@cthirouin-swi this was merely a optimisation to avoid running the parsing code but yet not treat this as an error ?
An empty response object do actually feel like an error to me

Reply:
You're right, I'll set an error for this.